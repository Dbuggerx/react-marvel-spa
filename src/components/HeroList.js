// @flow

import React from 'react';
import HeroCard from './HeroCard';
import type { Hero } from '../services/types';

type Props = {
    heroes: Hero[],
    getThumbnailUrl: (hero: Hero) => string,
};

const HeroesList = (props: Props) => (
    <div>
        {props.heroes.map(hero => (
            <HeroCard
                hero={hero}
                key={hero.id}
                imageUrl={props.getThumbnailUrl(hero)}
            />
        ))}
    </div>
);

HeroesList.displayName = 'HeroesList';

export default HeroesList;
