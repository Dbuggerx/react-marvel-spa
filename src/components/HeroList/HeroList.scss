@import '../scss/variables';

.hero-list {
  display: flex;
  flex-flow: row wrap;
  justify-content: flex-start;
  outline: none;

  &__card {
    width: 21%;
    margin: 2%;
    height: 11em;

    @media #{map-get($breakpoints, medium)} {
      width: 45%;
      margin: 2% 2.5%;
    }

    @media #{map-get($breakpoints, small)} {
      width: 100%;
      margin: 2% 0;
    }
  }
}
