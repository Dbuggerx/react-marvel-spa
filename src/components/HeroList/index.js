/* eslint-disable */

// @flow

import React, { Component } from 'react';
import HeroCard from '../HeroCard';
import type { Hero } from '../../services/types';

type Props = {
    heroes: Hero[],
    getThumbnailUrl: (hero: Hero) => string,
    onHeroClick: (hero: Hero) => void,
};

export default class HeroesList extends Component<Props> {
    render() {
        return this.props.heroes.map(hero => (
            <HeroCard
                onClick={this.props.onHeroClick}
                hero={hero}
                key={hero.id}
                imageUrl={this.props.getThumbnailUrl(hero)}
            />
        ));
    }
}
