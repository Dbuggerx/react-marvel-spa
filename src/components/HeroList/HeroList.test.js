import React from 'react';
import { shallow } from 'enzyme';
import HeroList from './index';

describe('HeroList', () => {
    let getThumbnailUrlMock;
    let handleHeroClickMock;
    let heroesMock;

    beforeEach(() => {
        getThumbnailUrlMock = jest.fn().mockImplementation(hero => `image URL for ${hero.name}`);
        handleHeroClickMock = jest.fn();
        heroesMock = [
            {
                id: 1,
                name: 'hero1',
            },
            {
                id: 2,
                name: 'hero2',
            },
        ];
    });

    describe('render', () => {
        test('renders properly with results', () => {
            const wrapper = shallow(<HeroList
                heroes={heroesMock}
                getThumbnailUrl={getThumbnailUrlMock}
                onHeroClick={handleHeroClickMock}
            />);
            expect(wrapper).toMatchSnapshot();
        });

        test('renders properly with no results', () => {
            const wrapper = shallow(<HeroList
                heroes={[]}
                getThumbnailUrl={getThumbnailUrlMock}
                onHeroClick={handleHeroClickMock}
            />);
            expect(wrapper).toMatchSnapshot();
        });
    });
});
