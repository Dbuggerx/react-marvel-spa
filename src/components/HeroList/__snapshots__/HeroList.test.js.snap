// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroList render renders properly with no results 1`] = `
<div
  className="hero-list"
  tabIndex="-1"
>
  <h2>
    No results found!
  </h2>
</div>
`;

exports[`HeroList render renders properly with results 1`] = `
<div
  className="hero-list"
  tabIndex="-1"
>
  <div
    className="hero-list__card"
    key="1"
  >
    <HeroCard
      hero={
        Object {
          "id": 1,
          "name": "hero1",
        }
      }
      imageUrl="image URL for hero1"
      onClick={[Function]}
    />
  </div>
  <div
    className="hero-list__card"
    key="2"
  >
    <HeroCard
      hero={
        Object {
          "id": 2,
          "name": "hero2",
        }
      }
      imageUrl="image URL for hero2"
      onClick={[Function]}
    />
  </div>
</div>
`;
